public/app/features/trails/MetricSelectScene.tsx

```diff
@@ -254,8 +256,8 @@ export class MetricSelectScene extends SceneObjectBase<MetricSelectSceneState> {
     }
   };
 
-  public onSearchChange = (evt: React.SyntheticEvent<HTMLInputElement>) => {
-    this.setState({ searchQuery: evt.currentTarget.value });
+  public onSearchChange = (searchQuery: string | undefined) => {
+    this.setState({ searchQuery });
     this.updateMetrics(); // Need to repeat entire pipeline
     this.buildLayout();
   };
```

Comment:
```ts
public onSearchChange = (evt: React.SyntheticEvent<HTMLInputElement>) => {
    this.setState({ searchQuery: evt.currentTarget.value });
    this.searchQueryChangedDebounced();
```

private searchQueryChangedDebounced = debounce(() => {
   this.updateMetrics()
    this.buildLayout();
}, 500);
```

I would do something like this to keep simpler (less need of new useState, useMemo, etc